Makefile

```diff
@@ -8,16 +8,25 @@ ns ?= "test"
 
 # Determine the operating system
 UNAME_S := $(shell uname -s)
+PHP_API_VERSION := $(shell php -i | grep 'PHP API' | awk -F' => ' '{print $$2}')
+PHP_VERSION := $(shell php -v | head -n 1 | awk '{print $$2}' | cut -d. -f1,2)
+
+# Check if PHP version is greater than 8.0
+ifeq ($(shell awk 'BEGIN{ print ("$(PHP_VERSION)" >= "8.0") }'), 0)
+    $(error PHP version must be greater than or equal to 8.0)
+endif
+
+
 ifeq ($(UNAME_S),Darwin)
     EXTENSION := .dylib
-	EXT_DIR_PATH := /opt/homebrew/opt/php@8.1/pecl/20210902
-    PHP_INI_PATH := /opt/homebrew/etc/php/8.1/php.ini
-    RESTART_COMMAND := brew services restart php@8.1
+	EXT_DIR_PATH := /opt/homebrew/opt/php@$(PHP_VERSION)/pecl/$(PHP_API_VERSION)
```

Comment:
Are you assuming that PHP is preinstalled in this path? Do we have enough documentation to assume this dependency will be met on install?

Reply:
yes for macOS this is the default installation path. There is documentation on homebrew as well confirming this.